core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -164,13 +164,11 @@ class BrokerToControllerRequestThread(networkClient: KafkaClient,
                                       listenerName: ListenerName,
                                       time: Time,
                                       threadName: String)
-  extends InterBrokerSendThread(threadName, networkClient, time, isInterruptible = false) {
+  extends InterBrokerSendThread(threadName, networkClient, config.controllerSocketTimeoutMs, time, isInterruptible = false) {
 
   private var activeController: Option[Node] = None
 
-  override def requestTimeoutMs: Int = config.controllerSocketTimeoutMs
-
-  override def generateRequests(): Iterable[RequestAndCompletionHandler] = {
+  def generateRequests(): Iterable[RequestAndCompletionHandler] = {
```

Comment:
Not needed here, but I wonder if we should just make this return an `Option[RequestAndCompletionHandler]`

Reply:
Can you clarify what the option would achieve over an empty collection?

Reply:
I was just noticing that we only ever return a Queue with a single item. Optional seems to fit that case more cleanly. However (as we discussed offline), I wonder why we have the code like this in the first place. Maybe @abbccdda has some insight?

Reply:
Yeah, I agree that looks a bit odd. It looks like it was intended to be a loop.